dace/codegen/targets/xilinx.py

```diff
@@ -802,9 +840,11 @@ def generate_kernel_internal(self, sdfg: dace.SDFG, state: dace.SDFGState, kerne
                                                entry_stream)
 
         for is_output, name, node, _ in external_streams:
+            # TODO find better way to extract num kernels, rather than the shape of the streams
+            num_kernels = dace.symbolic.evaluate(node.shape[0], sdfg.constants)
```

Comment:
Didn't you already put this `n_unrolled` thing?

Reply:
Only in `rtl.py`, and this is in `xilinx.py`. However, I'm thinking this TODO should just be removed, since this is only used for generating the configuration file, which does not allow for array of streams. So array of streams should be unrolled in the configuration file. Maybe it should be renamed, since there doesn't necessarily have to be multiple kernels? So something like `num_streams`?

Reply:
Okay, if you're getting `num_streams` then this line makes perfect sense :-)

Reply:
This has been changed.